app/controllers/api/v1/base_controller.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class BaseController < ActionController::API
+      include CanCan::ControllerAdditions
+      include RequestTimeouts
+      include Pagy::Backend
+      include JsonApiPagination
+
+      check_authorization
+
+      attr_accessor :current_api_user
+
+      before_action :authenticate_user
+
+      rescue_from Exception, with: :error_during_processing
+      rescue_from CanCan::AccessDenied, with: :unauthorized
+      rescue_from ActiveRecord::RecordNotFound, with: :not_found
+
+      private
+
+      def authenticate_user
+        return if (@current_api_user = request.env['warden'].user)
+
+        if api_key.blank?
+          # An anonymous user
+          @current_api_user = Spree::User.new
+          return
+        end
+
+        return if (@current_api_user = Spree::User.find_by(spree_api_key: api_key.to_s))
+
+        invalid_api_key
+      end
+
+      def current_ability
+        Spree::Ability.new(current_api_user)
+      end
+
+      def api_key
+        request.headers["X-Api-Token"] || params[:token]
+      end
+
+      def error_during_processing(exception)
+        Bugsnag.notify(exception)
+
+        render status: :unprocessable_entity,
+               json: json_api_error(exception.message, backtrace: exception.backtrace)
```

Comment:
Are we sure we want to publish the message? It could expose sensitive information.

Reply:
It only gets shown in development environment (in the `#json_api_error` method).

Reply:
The backtrace was hidden in production but the message was still passed on. I added a commit to my branch:
https://github.com/openfoodfoundation/openfoodnetwork/commit/7578085ec7af951b1ff11c96fe351e98a966d9d7